src/Microsoft.ML.FastTree/TreeEnsemble/RegressionTree.cs

```diff
@@ -14,25 +14,73 @@
 using Microsoft.ML.Model;
 using Microsoft.ML.Model.Pfa;
 using Newtonsoft.Json.Linq;
-using Float = System.Single;
 
 namespace Microsoft.ML.Trainers.FastTree.Internal
 {
-    public class RegressionTree
+    public class RegressionTreeView
```

Comment:
>RegressionTreeView [](start = 17, length = 18)

From an external perspective this name is problematic. Sure you've architectured it as a wrapper, but the name "view" asks people to wonder, a view of what? If that regression tree below were public you might have a case, but that is something we want to hide. I don't think you're going to get out of this without changing the name of the internal thing.

Reply:
How about `DecisionTreeRegressor` for this public class (inspired by ONNX)?

---
In reply to: [251182801](https://github.com/dotnet/machinelearning/pull/2243#discussion_r251182801) [](ancestors = 251182801)

Reply:
But it doesn't perform regression, so calling it a `Regressor` makes no sense? How about just `DecisionTree`?